ddtrace/contrib/graphql/patch.py

```diff
@@ -0,0 +1,279 @@
+import os
+import re
+import sys
+from typing import TYPE_CHECKING
+
+import graphql
+from graphql import MiddlewareManager
+from graphql.error import GraphQLError
+from graphql.execution import ExecutionResult
+from graphql.language.source import Source
```

Comment:
Some thoughts for potential follow-up work: use `ModuleWatchdog` to detect when the `graphql` module is imported to defer these imports to after that happens, as part of the `patch` logic. Hopefully, this way we could save on startup time (and potentially on the memory footprint if the same is applied to other integrations that do eager imports of modules that might never be used).

Reply:
It is likely worth an investigation into using `ModuleWatchdog` for all integrations. We always wanted to make all integrations "patch on import", but previous attempt have been lacking. `ModuleWatchdog` is probably best we've had so far.